libvast/src/system/transformer.cpp

```diff
@@ -39,13 +39,20 @@ transformer_stream_stage_ptr attach_transform_stage(
         self->send_exit(self, caf::make_error(ec::end_of_input));
         return;
       }
-      auto transformed = self->state.transforms.apply(std::move(x.body));
+      auto add_failed = self->state.transforms.add(std::move(x.body));
+      if (add_failed) {
+        VAST_ERROR("add failed {}", add_failed);
+        return;
```

Comment:
Other errors above cause the transformer to exit, why is this not the case here?

Reply:
The above is not an error case. Here this is an error case and the transformer skips the slice but continues the transformation with the rest of the slices.

Reply:
If this is not an error, then why does this use `VAST_ERROR` rather than `VAST_WARN`?

Reply:
To be more precise:

```
      if (x.header == detail::stream_control_header::eof) {
        VAST_DEBUG("{} quits after receiving EOF control message in stream",
                   self->state.transformer_name);
        self->send_exit(self, caf::make_error(ec::end_of_input));
        return;
      }
      auto add_failed = self->state.transforms.add(std::move(x.body));
      if (add_failed) {
        VAST_ERROR("add failed {}", add_failed);
        return;
      }
```

The first `if` is not an error case. The second `if` is an error case, in this case the transformer skips the slice but continues the transformation with the rest of the slices. In this situation I believe that `VAST_WARN` can also be used. @lava  What is your opinion?